### ISSUE TITLE
[Bug]: SSL errors on MacOS localhost connection even after specifying ignoreHTTPSErrors: true

### ISSUE THREAD

**Issue by romka-best** [microsoft/playwright#32304]
### Version

1.46.1

### Steps to reproduce

1. Start a web server locally on port 3000
2. Run the test

### Expected behavior

The localhost:3000 link should open a test web page currently being served

### Actual behavior

Failed to load resource: An SSL error has occurred and a secure connection to the server cannot be made.
<img width="1275" alt="Screenshot 2024-08-23 at 2 12 36 PM" src="https://github.com/user-attachments/assets/53c6541a-998a-498b-9e59-cf4c08d94966">

### Environment

```shell
System:
    OS: macOS 14.6.1
    CPU: (8) arm64 Apple M1 Pro
    Memory: 73.23 MB / 16.00 GB
  Binaries:
    Node: 18.20.4 - ~/.nvm/versions/node/v18.20.4/bin/node
    Yarn: 1.22.19 - /usr/local/bin/yarn
    npm: 10.7.0 - ~/.nvm/versions/node/v18.20.4/bin/npm
    pnpm: 9.7.1 - ~/.nvm/versions/node/v18.20.4/bin/pnpm
  IDEs:
    VSCode: 0.39.6 - /usr/local/bin/code
  Languages:
    Bash: 3.2.57 - /bin/bash
  npmPackages:
    @playwright/test: 1.46.1 => 1.46.1
```

**Comment by mxschmitt** [comment]
Could you share your test file and config with us?

**Comment by romka-best** [comment]
> Could you share your test file and config with us?

Yes, of course

Config:
```
import { defineConfig, devices } from '@playwright/test';

/**
 * Read environment variables from file.
 * https://github.com/motdotla/dotenv
 */
import dotenv from 'dotenv';
import path from 'path';
dotenv.config({ path: path.resolve(__dirname, '.env') });

/**
 * See https://playwright.dev/docs/test-configuration.
 */
export default defineConfig({
  testDir: './tests',
  snapshotPathTemplate: '{testDir}/{testFileDir}/__screenshots__/{arg}-{projectName}{ext}',
  /* Run tests in files in parallel */
  fullyParallel: true,
  /* Fail the build on CI if you accidentally left test.only in the source code. */
  forbidOnly: !!process.env.CI,
  /* Retry on CI only */
  retries: process.env.CI ? 2 : 0,
  /* Opt out of parallel tests on CI. */
  workers: process.env.CI ? 1 : undefined,
  /* Timeout for each test */
  // 2 minutes.
  timeout: 2 * 60 * 1000,
  /* Reporter to use. See https://playwright.dev/docs/test-reporters */
  reporter: process.env.CI ? 'blob' : 'html',
  expect: {
    timeout: 10 * 1000,
  },
  /* Shared settings for all the projects below. See https://playwright.dev/docs/api/class-testoptions. */
  use: {
    actionTimeout: 10 * 1000,

    /* Collect trace when retrying the failed test. See https://playwright.dev/docs/trace-viewer */
    trace: process.env.CI ? 'on-first-retry' : 'retain-on-failure',
  },

  /* Configure projects for major browsers */
  projects: [
    /* Test against desktop browsers */
    {
      name: 'Google Chrome',
      use: { ...devices['Desktop Chrome'] },
    },
    {
    name: 'Firefox',
    use: { ...devices['Desktop Firefox'] },
    },
    {
    name: 'Safari',
    use: { ...devices['Desktop Safari'] },
    },

    /* Test against mobile viewports */
    {
    name: 'Mobile Chrome',
    use: { ...devices['Pixel 5'] },
    },
    {
    name: 'Mobile Safari',
    use: { ...devices['iPhone 12'] },
    },
  ],
});
```

Test file:
```
import { expect, test } from '@playwright/test';

import { serverTOSURL } from '../common/def';

const TERMS_OF_SERVICE_SELECTOR = '.t_terms_of_service';

test.describe('[Terms of Service]', () => {
  test.beforeEach(async ({ page }) => {
    await page.setExtraHTTPHeaders({
      'x-vercel-skip-toolbar': '1',
    });

    await page.goto(serverTOSURL);
  });

  test('Content', async ({ page }) => {
    await page.waitForSelector(TERMS_OF_SERVICE_SELECTOR);

    await expect(page).toHaveScreenshot({ maxDiffPixelRatio: 0.1 });
  });
});
```

**Comment by dgozman** [comment]
Closing because we lack the information to act on this issue. If you can provide a full repro, please file a new issue by filling the "Bug Report" template, and link to this one.
